src/utils/audit-trail/user.ts

```diff
@@ -0,0 +1,53 @@
+import { Request } from 'express';
+
+import { UserQuery } from '../__generated__/graphql-request';
+import { isAuditTrailEnabledForOrg, workos } from './workos';
+import { logger } from '@/logger';
+
+export async function createAuditLog(
+  action: string,
+  user: UserQuery['user'],
+  req: Request,
+  versionId = 1
+) {
+  return user?.organization_users.map((organizationUser) => {
+    if (
+      organizationUser.disabled ||
+      !isAuditTrailEnabledForOrg(organizationUser.organization)
+    ) {
+      return;
+    }
+
+    try {
+      return workos.auditLogs.createEvent(
```

Comment:
```suggestion
      return await workos.auditLogs.createEvent(
```

Have to `await` here, otherwise rejection will not be caught.